go/vt/vttablet/tabletserver/query_executor.go

```diff
@@ -245,6 +274,39 @@ func (qre *QueryExecutor) Stream(callback func(*sqltypes.Result) error) error {
 		return err
 	}
 
+	sql, sqlWithoutComments, err := qre.generateFinalSQL(qre.plan.FullQuery, qre.bindVars)
+	if err != nil {
+		return err
+	}
+
+	var replaceKeyspace string
+	if sqltypes.IncludeFieldsOrDefault(qre.options) == querypb.ExecuteOptions_ALL {
+		replaceKeyspace = qre.tsv.sm.target.Keyspace
+	}
+
+	if consolidator := qre.tsv.qe.streamConsolidator; consolidator != nil {
+		if qre.connID == 0 && qre.plan.PlanID == p.PlanSelectStream && qre.shouldConsolidate() {
+			return consolidator.Consolidate(qre.logStats, sqlWithoutComments, callback,
+				func(callback StreamCallback) error {
+					dbConn, err := qre.getStreamConn()
+					if err != nil {
+						return err
+					}
+					defer dbConn.Recycle()
+					return qre.execStreamSQL(dbConn, sql, func(result *sqltypes.Result) error {
+						// this stream result is potentially used by more than one client, so
+						// the consolidator will return it to the pool once it knows it's no longer
+						// being shared
+
+						if replaceKeyspace != "" {
+							result.ReplaceKeyspace(replaceKeyspace)
+						}
+						return callback(result)
```

Comment:
replace keyspace does not need to called always as not all queries will be for keyspace tables. It can be a information_schema query or _vt query.

There is similar check in `Execute` method where we check for if dbname is different from keyspace name and also if field.Database == dbname then only replace with the keyspace name.

Reply:
@harshit-gangal this is the same behavior that we had before the PR; I've just refactored it by lifting it from the caller. I'm pretty sure we're only replacing the keyspace when it's required (see https://github.com/vitessio/vitess/pull/7752/files#diff-799d4762d9e64026cb623db2defeb53209df055a99c1106f535006a89e7b812aR282-R285, which I've lifted from https://github.com/vitessio/vitess/pull/7752/files#diff-28fc9df9d28d69ca625b85e18a6fa652d61bfaf6de03012ce2d8d0338363d70eL808-L817) Can you double-check?

Reply:
(unless the behavior before the PR was already broken, in that case I'll gladly fix it now :smile:)

Reply:
I realised that, it was broken before this PR